### PR TITLE
Add helm framework to tests

### PR DESCRIPTION
**Pull request by rohitsakala**
Adding helm framework to tests so that integration or validation tests which require doing `helm install rancher` can use this framework.

### CODE REVIEW

File: tests/framework/clients/helm/helm.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var helmCmd = "helm_v3"
+
+func InstallRancher(restConfig *rest.Config) error {
```
Comment (igomez06) [inline tests/framework/clients/helm/helm.go:16]: Can we get comments on all the export function? For our godocs

**Review by igomez06 — COMMENTED** [review]
Add comments please. Otherwise looks good. I want at least one more person to review as well.

File: tests/framework/clients/helm/helm.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	err = InstallChart("cert-manager", "jetstack/cert-manager", "cert-manager", "", "--set", "installCRDs=true")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Install a helm chart
+func InstallChart(releaseName, helmRepo, namespace, version string, args ...string) error {
```
Comment (caliskanugur) [inline tests/framework/clients/helm/helm.go:92]: I am sure I am missing context about it, and the answer probably depends on the plan and how the container is going to be deleted; would it make sense to add a session to bind an uninstall chart action in this function? 

Sessions allow us to register different functions -- they are usually used for delete/uninstall calls. We can think about them as queues. The session cleanup can be called either in teardown or at the end of the tests, wherever we see the best fit.

**Review by caliskanugur — COMMENTED** [review]
Looks good to me; I've got two questions that purely depend on the plan for this client

File: tests/framework/clients/helm/helm.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	// Install cert-manager Chart
+	err = InstallChart("cert-manager", "jetstack/cert-manager", "cert-manager", "", "--set", "installCRDs=true")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/framework/clients/helm/helm.go:89]: I agree with this, I think this kind follows the structure of k3d and corral

File: tests/framework/clients/helm/helm.go (hunk 2 above)
Comment (igomez06) [reply tests/framework/clients/helm/helm.go:92]: Something like this Rohit: https://github.com/rancher/rancher/blob/4367bc51f4de4c59250e7b20ddf2a1382dab04e4/tests/framework/clients/k3d/k3d.go#L34-L36

File: tests/framework/clients/helm/helm.go (hunk 1 above)
Comment (rohitsakala) [reply tests/framework/clients/helm/helm.go:22]: Done.

File: tests/framework/clients/helm/helm.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rohitsakala) [reply tests/framework/clients/helm/helm.go]: Done. Separated them into another file

File: tests/framework/clients/helm/helm.go (hunk 2 above)
Comment (rohitsakala) [reply tests/framework/clients/helm/helm.go]: Added the registerCleanup for both Install and Upgrade charts.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Review by igomez06 — APPROVED** [review]
LGTM
please create a backport. Thank you!

**Pull request merged**